### ISSUE TITLE
[FR] implement webhooks for common git hosting sites

### ISSUE THREAD

**Issue by blaggacao** [rancher/terraform-operator#20]
Role model: https://www.runatlantis.io/

**Comment by cloudnautique** [comment]
Yeah, this is certainly makes sense as a general purpose TF runner.  Our initial usecases are more limited to some very specific and controlled models. 

One thought I have had is to layer Argo and Argo events on top of this. That would provide a few things like a UI, workflow and eventing.
